diff --git a/build-scripts/config.sub b/build-scripts/config.sub
index 5ecc18b..f9f0115 100755
--- a/build-scripts/config.sub
+++ b/build-scripts/config.sub
@@ -1297,7 +1297,8 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
+	      | -helenos* )
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff --git a/configure.in b/configure.in
index 08c8e1e..337d985 100644
--- a/configure.in
+++ b/configure.in
@@ -2494,6 +2494,15 @@ case "$host" in
             have_timers=yes
         fi
         ;;
+    *-*-helenos*)
+        ARCH=helenos
+        CheckDummyVideo
+        CheckDiskAudio
+        CheckDummyAudio
+        # CheckNASM
+        CheckDLOPEN
+        CheckPTHREAD
+        ;;
     *-*-cygwin* | *-*-mingw32*)
         ARCH=win32
         if test "$build" != "$host"; then # cross-compiling
