diff --git a/config.sub b/config.sub
index dd2ca93..79172bf 100755
--- a/config.sub
+++ b/config.sub
@@ -1409,7 +1409,7 @@ case $os in
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -helenos* )
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff --git a/include/cross.h b/include/cross.h
index 5f941a6..3097585 100644
--- a/include/cross.h
+++ b/include/cross.h
@@ -64,7 +64,7 @@
 //Solaris maybe others
 #if defined (DB_HAVE_NO_POWF)
 #include <math.h>
-static inline float powf (float x, float y) { return (float) pow (x,y); }
+//static inline float powf (float x, float y) { return (float) pow (x,y); }
 #endif
 
 class Cross {
diff --git a/include/setup.h b/include/setup.h
index 35e386d..576bfc5 100644
--- a/include/setup.h
+++ b/include/setup.h
@@ -108,9 +108,9 @@ public:
 	std::string ToString() const;
 
 private:
-	void destroy() throw();
+	void destroy()/* throw()*/;
 	Value& copy(Value const& in);
-	void plaincopy(Value const& in) throw();
+	void plaincopy(Value const& in)/* throw()*/;
 	void set_hex(std::string const& in);
 	void set_int(std::string const&in);
 	void set_bool(std::string const& in);
diff --git a/src/cpu/cpu.cpp b/src/cpu/cpu.cpp
index 363b091..8264843 100644
--- a/src/cpu/cpu.cpp
+++ b/src/cpu/cpu.cpp
@@ -22,7 +22,7 @@
 #include <sstream>
 #include "dosbox.h"
 #include "cpu.h"
-#include "memory.h"
+#include <memory>
 #include "debug.h"
 #include "mapper.h"
 #include "setup.h"
diff --git a/src/dos/cdrom_image.cpp b/src/dos/cdrom_image.cpp
index bdf41b2..dad8f02 100644
--- a/src/dos/cdrom_image.cpp
+++ b/src/dos/cdrom_image.cpp
@@ -19,7 +19,7 @@
 /* $Id: cdrom_image.cpp,v 1.24 2009-03-19 20:45:42 c2woody Exp $ */
 
 #include <cctype>
-#include <cmath>
+//#include <cmath>
 #include <cstdio>
 #include <fstream>
 #include <iostream>
@@ -33,6 +33,7 @@
 #include "support.h"
 #include "setup.h"
 
+#define WIN32
 #if !defined(WIN32)
 #include <libgen.h>
 #else
diff --git a/src/dosbox.cpp b/src/dosbox.cpp
index edee31a..fde2250 100644
--- a/src/dosbox.cpp
+++ b/src/dosbox.cpp
@@ -47,6 +47,9 @@ Config * control;
 MachineType machine;
 SVGACards svgaCard;
 
+// XXX Just a workaround for HelenOS build failure
+void *__dso_handle = 0;
+
 /* The whole load of startups for all the subfunctions */
 void MSG_Init(Section_prop *);
 void LOG_StartUp(void);
diff --git a/src/gui/sdlmain.cpp b/src/gui/sdlmain.cpp
index 0704c9b..59d2bee 100644
--- a/src/gui/sdlmain.cpp
+++ b/src/gui/sdlmain.cpp
@@ -2049,7 +2049,7 @@ int main(int argc, char* argv[]) {
 		control->StartUp();
 		/* Shutdown everything */
 	} catch (char * error) {
-		GFX_ShowMsg("Exit to error: %s",error);
+//		GFX_ShowMsg("Exit to error: %s",error);
 		fflush(NULL);
 		if(sdl.wait_on_error) {
 			//TODO Maybe look for some way to show message in linux?
diff --git a/src/hardware/mame/emu.h b/src/hardware/mame/emu.h
index 2d1fef9..77c8aaf 100644
--- a/src/hardware/mame/emu.h
+++ b/src/hardware/mame/emu.h
@@ -11,7 +11,7 @@
 #include <math.h>
 #include <float.h>
 #include <stdlib.h>
-#include <memory.h>
+#include <cstring>
 
 #if C_DEBUG
 #include <stdio.h>
diff --git a/src/misc/setup.cpp b/src/misc/setup.cpp
index 2cb0d1c..de292c7 100644
--- a/src/misc/setup.cpp
+++ b/src/misc/setup.cpp
@@ -32,7 +32,7 @@
 
 using namespace std;
 static std::string current_config_dir; // Set by parseconfigfile so Prop_path can use it to construct the realpath
-void Value::destroy() throw(){
+void Value::destroy() /*throw()*/{
 	if (type == V_STRING) delete _string;
 }
 
@@ -45,7 +45,7 @@ Value& Value::copy(Value const& in) {
 	return *this;
 }
 
-void Value::plaincopy(Value const& in) throw(){
+void Value::plaincopy(Value const& in) /*throw()*/{
 	type = in.type;
 	_int = in._int;
 	_double = in._double;
@@ -814,7 +814,7 @@ bool Config::ParseConfigFile(char const * const configfilename){
 			try {
 				if(currentsection) currentsection->HandleInputline(gegevens);
 			} catch(const char* message) {
-				message=0;
+				//message=0;
 				//EXIT with message
 			}
 			break;
diff --git a/src/shell/shell_cmds.cpp b/src/shell/shell_cmds.cpp
index 1aa2e80..f393ec7 100644
--- a/src/shell/shell_cmds.cpp
+++ b/src/shell/shell_cmds.cpp
@@ -973,11 +973,11 @@ void DOS_Shell::CMD_SUBST (char * args) {
 		this->ParseLine(mountstring);
 	}
 	catch(int a){
-		if(a == 0) {
+//		if(a == 0) {
 			WriteOut(MSG_Get("SHELL_CMD_SUBST_FAILURE"));
-		} else {
-		       	WriteOut(MSG_Get("SHELL_CMD_SUBST_NO_REMOVE"));
-		}
+//		} else {
+//		       	WriteOut(MSG_Get("SHELL_CMD_SUBST_NO_REMOVE"));
+//		}
 		return;
 	}
 	catch(...) {		//dynamic cast failed =>so no localdrive
